reference/conanfile/methods/build_requirements.rst

```diff
@@ -0,0 +1,52 @@
+.. _reference_conanfile_methods_build_requirements:
+
+build_requirements()
+====================
+
+The ``build_requirements()`` method is functionally equivalent to the ``requirements()`` one, it is executed just after it. It is not strictly necessary, in theory everything that is inside this method, could be done in the end of the ``requirements()`` one. Still, the ``build_requirements()`` is good for having a dedicated place to define ``tool_requires`` and ``test_requires``:
```

Comment:
can 2.0 rename it to `tool_requirements`? it's kinda weird and confusing different places use different terminology

Reply:
No, it needs to keep compatibility with 1.X, which has not renamed the  method, and it is too late to do it.

Reply:
why does it need? 2.0 may have breaking changes, it's major bump.

Reply:
Conan 2.0 keeps full compatibility with some sub-syntax of 1.X, otherwise the migration wouldn't be possible. ``build_requirements()`` is in that subset of the syntax that is necessary for compatibility with 1.X